python3Packages.ipfshttpclient: 0.4.2.post1 -> 0.7.0, rename from ipfsapi

Luflosi opened the pull request:
###### Motivation for this change
This adds ipfshttpclient from https://github.com/ipfs-shipyard/py-ipfs-http-client, which has been renamed from ipfsapi.
The patch changing `VERSION_MAXIMUM` is needed because this library hardcodes a maximum version of IPFS, which is older then the version currently in nixpkgs. It seems to work just fine though.
I think completely removing the old ipfsapi package is fine since that version is from 2017. And while a compatibility layer has been written to make migrations easier, it stopped getting updates at the end of 2019 according to https://github.com/ipfs-shipyard/py-ipfs-http-client/blob/910d11f87f1e22a1d07a684a2ca3852841e07bd7/ipfsapi/__init__.py#L8 and also only works with versions 0.4.x of this library, not 0.5.x or even 0.6.x. To package the compatibility layer, we would need to either patch it to work with the latest version or package an outdated version of this package as well.
The tests start an IPFS daemon, which listens on some ports. The build is usually sandboxed and without internet connection, so this should be perfectly fine, right?
I kept the original maintainer @mguentner, even though he hasn't updated the package since 2017.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Luflosi committed:
python3Packages.ipfshttpclient: 0.4.2.post1 -> 0.7.0, rename from ipfsapi

r-rmcgibbo commented:
Result of `nixpkgs-review pr 115984` at 20b4d704 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li><a href="https://gist.github.com/a5936679a1490aa591def1e2c869da86">python38Packages.ipfshttpclient</a></li></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `python3`.
Near pkgs/development/python-modules/ipfshttpclient/default.nix:5:3:
```
  |
5 | , python3
  |   ^
```
</li>
  </ul>
</details>

Luflosi commented:
I removed the unused argument.

SuperSandro2000 commented on pkgs/development/python-modules/ipfshttpclient/default.nix:
```diff
@@ -0,0 +1,75 @@
+{ lib
+, buildPythonPackage
```
```suggestion
, buildPythonPackage
, python
```

SuperSandro2000 commented on pkgs/development/python-modules/ipfshttpclient/default.nix:
```diff
@@ -0,0 +1,75 @@
+  checkPhase = ''
+    runHook preCheck
+
+    python3 -X utf8 test/run-tests.py
```
```suggestion
    ${python.interpreter} -X utf8 test/run-tests.py
```

Otherwise it might break with pypy or python2 (theoretically).

Luflosi commented:
Thank you, I implemented your suggestion.

SuperSandro2000 commented:
13 tests fail with 

```
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

tests_content = '\n    import pytest\n\n    @pytest.mark.run(order=-1)\n    class TestLast(object):\n\n        def test_1(self): pass\....run(order=-2)\n    class TestFirst(object):\n\n        def test_4(self): pass\n\n        def test_5(self): pass\n    '

    def _item_names_for(tests_content):
        # some strange code to extract sorted items
        items = testdir.getitems(tests_content)
>       hook = testdir.config.hook
E       AttributeError: 'Testdir' object has no attribute 'config'

/build/source/tests/test_ordering.py:15: AttributeError
----------------------------- Captured stdout call -----------------------------
```

Luflosi commented:
Apparently the dependency pytest-ordering fails to compile, I'll see what I can do.

Luflosi commented:
I replaced pytest-ordering with  pytest-order, which is a maintained fork.
I also updated ipfshttpclient to 0.7.0, which is the latest version.
Please review.

Luflosi commented:
Thank you for updating the title of this PR to reflect the change in the commit. I forgot to change it.
